### DISCUSSION TITLE
[TypeScript] Add `undefined` to Atom definition

### DISCUSSION THREAD

**Discussion by rtritto** [pmndrs/jotai#2667] [General]
### Actual
With
```ts
const myAtom = atom<string>()
```
in VSCode I get
```console
Expected 1 arguments, but got 0.ts(2554)
atom.d.ts(38, 37): An argument for 'read' was not provided
```

---

Instead, as workaround with
```ts
const myAtom = atom<string | undefined>(undefined)
```

in ESLint I get
```console
error  Do not use useless `undefined`  unicorn/no-useless-undefined
```

### Expected
No Error in VSCode with
```ts
const myAtom = atom<string>()
```
because Atom definition should already use `undefined` as default value

### Related example (React)
```ts
const [myState, setMyState] = useState<string>()
```

**Comment by dai-shi** [comment]
Thanks for your suggestion.
I'm aware of the limitation.

My workaround has been `atom<string | undefined>(undefined)` or `atom(undefined as string | undefined)`.

I think adding one more type overload should fix. Would you like to open a PR?

On the other hand, the rationale behind the current design decision is `atom<string>()` may lead bugs for users, because the `undefined` type is implicit (only inferred). I wonder how others care about it.

**Reply by rtritto** [reply]
I opened #2668
